capstone/src/main/java/com/google/sps/gmail/EmailFactory.java

```diff
@@ -0,0 +1,88 @@
+package com.google.sps.gmail;
+
+import com.google.api.services.gmail.Gmail;
+import com.google.api.services.gmail.model.Message;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Streams;
+import com.google.sps.servlets.Constants;
+import com.google.sps.servlets.ServletUtil;
+import com.google.sps.servlets.StudentServlet;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Properties;
+import javax.mail.Message.RecipientType;
+import javax.mail.MessagingException;
+import javax.mail.Session;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+import org.apache.commons.codec.binary.Base64;
+
+public class EmailFactory {
```

Comment:
Nice job on the factory class! Can you add tests for these new classes?

Reply:
Done!